src/chunk_constraint.c

```diff
@@ -468,10 +469,19 @@ ts_chunk_constraint_scan_by_dimension_slice(DimensionSlice *slice, ChunkScanCtx
 
 		ts_hypercube_add_slice(chunk->cube, slice);
 
-		if (ctx->early_abort && chunk->constraints->num_dimension_constraints == hs->num_dimensions)
+		/* A chunk is complete when we've added slices for all its dimensions,
+		 * i.e., a complete hypercube */
+		chunk_is_complete = chunk->constraints->num_dimension_constraints == hs->num_dimensions;
```

Comment:
There are static function `chunk_is_complete()` exist in `chunk.c`. Maybe it is better to make it public and use here instead? If that makes sense

Reply:
Made it static inline in the header.